lockbox-ios/View/AutoLockSettingsView.swift

```diff
@@ -0,0 +1,101 @@
+
+class AutoLockSettingsView: UITableViewController {
+    var presenter: AutoLockSettingsPresenter?
+    private var disposeBag = DisposeBag()
+    private var dataSource: RxTableViewSectionedReloadDataSource<AutoLockSettingSectionModel>?
+
+    init() {
+        super.init(nibName: nil, bundle: nil)
+        self.presenter = AutoLockSettingsPresenter(view: self)
+        view.backgroundColor = Constant.color.settingsBackground
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return UIStatusBarStyle.lightContent
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupNavbar()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupDataSource()
+        setupDelegate()
+        presenter?.onViewReady()
+    }
+
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let cell = UITableViewCell()
+        cell.textLabel?.textColor = Constant.color.settingsHeader
+        cell.textLabel?.font = UIFont.systemFont(ofSize: 13.0, weight: UIFont.Weight.regular)
+        cell.textLabel?.text = Constant.string.autoLockHeader
+        cell.textLabel?.textAlignment = NSTextAlignment.center
+        return cell
+    }
+}
+
+extension AutoLockSettingsView {
+    private func setupNavbar() {
+        navigationItem.title = Constant.string.settingsAutoLock
+    }
+
+    private func setupDataSource() {
+        dataSource = RxTableViewSectionedReloadDataSource(
+            configureCell: {(_, _, _, cellConfiguration) -> UITableViewCell in
+            let cell = UITableViewCell()
+            cell.textLabel?.text = cellConfiguration.text
+
+            cell.accessoryType = cellConfiguration.isChecked ?
+                UITableViewCellAccessoryType.checkmark : UITableViewCellAccessoryType.none
+
+            return cell
+        })
+    }
+
+    private func setupDelegate() {
+        tableView.rx.modelSelected(CheckmarkSettingCellConfiguration.self)
+            .subscribe(onNext: { model in
+                guard let val = model.value as? AutoLockSetting,
+                      let presenter = self.presenter else { return }
+
+                if let indexPath = self.tableView.indexPathForSelectedRow {
+                    self.tableView.deselectRow(at: indexPath, animated: true)
+                }
+
+                presenter.itemSelected(val)
+            }).disposed(by: self.disposeBag)
+    }
+}
+
+extension AutoLockSettingsView: AutoLockSettingsProtocol {
+    func bind(items: SharedSequence<DriverSharingStrategy, [AutoLockSettingSectionModel]>) {
+        guard let dataSource = self.dataSource else {
+            fatalError("datasource not set!")
+        }
+
+        items
+            .drive(self.tableView.rx.items(dataSource: dataSource))
```

Comment:
`self.dataSource`

Reply:
I think this use of `dataSource` without `self` is because it is defined in the `guard` line above

Reply:
you're right, my B